Add ChIP ingress adapter for OTI telemetry

pkcll opened the pull request:
Implements ChipIngressAdapter to forward OTI telemetry to Beholder using chain-selectors for network/chain metadata derivation.

- Add ChipIngressAdapter with chain selector derivation
- Map TelemetryType to Beholder domain/entity pairs
- Emit telemetry with chain metadata attributes

JIRA: [INFOPLAT-2731](https://smartcontract-it.atlassian.net/browse/INFOPLAT-2731)



[INFOPLAT-2731]: https://smartcontract-it.atlassian.net/browse/INFOPLAT-2731?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hendoxc commented on core/services/telemetry/chip_ingress.go:
```diff
@@ -0,0 +1,152 @@
+var _ commontypes.MonitoringEndpoint = (*ChipIngressAdapter)(nil)
+
+// Emitter is the interface for the beholder.Emitter
+// https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/beholder/client.go#L27
+type Emitter interface {
+	Emit(ctx context.Context, body []byte, attrKVs ...any) error
+}
```
do we maybe want to make this `Emit` + `EmitSync` ?

pkcll replied:
The interface is identical to what is implemented in chainlink-common Emitter. Currently its only Emit. 
Adding EmitSync would require implementing it for chainlink-common Emitter.

pkcll committed:
feat: add ChIP ingress adapter for OTI telemetry

Implements ChipIngressAdapter to forward OTI telemetry to Beholder
using chain-selectors for network/chain metadata derivation.

- Add ChipIngressAdapter with chain selector derivation
- Map OTI TelemetryType to Beholder domain/entity pairs
- Emit telemetry with chain metadata attributes

JIRA: INFOPLAT-2731

pavel-raykov commented on core/services/telemetry/chip_ingress.go:
```diff
@@ -0,0 +1,152 @@
+// telemTypeToDomainAndEntity maps TelemetryType to (domain, entity) pairs for beholder
+// This function is based on the mapping from:
+// https://github.com/smartcontractkit/atlas/blob/e0dfd7dbd28fc79890e8d0bcae6b9c8eddfba01b/ingress/ocr-telemetry/app/chip_ingress_batcher.go#L232-L278
+func telemTypeToDomainAndEntity(telemType synchronization.TelemetryType) (domain, entity string, err error) {
```
would it make sense to put such common stuff into https://github.com/smartcontractkit/chainlink-common/tree/main/pkg/beholder ?

pkcll replied:
This mapping is only related to OTI to chip-ingress migration. It will not be used anywhere else

4of9 commented on core/services/telemetry/chip_ingress.go:
```diff
@@ -0,0 +1,152 @@
+		"beholder_domain", a.Domain,
+		"beholder_entity", a.Entity,
+		"chain_id", a.ChainID,
+		"network", a.Network,
```
should we call this `network_name` to be a bit more descriptive?

hendoxc commented on core/services/telemetry/chip_ingress.go:
```diff
@@ -0,0 +1,152 @@
+// SendLog implements commontypes.MonitoringEndpoint
+// It forwards the telemetry log to the beholder emitter with proper domain/entity attributes
+func (a *ChipIngressAdapter) SendLog(log []byte) {
+	ctx := context.Background()
```
do we need a ctx with timeout of like `10s` ?

pkcll replied:
Not need to use context.WithTimeout because emitter.Emit is async and uses context.WithoutCancel(ctx)
See https://github.com/smartcontractkit/chainlink-common/pull/1442/files#diff-325f5f30327d23d72937e111f12775ab7a63454159a8d6748ec20bf83d944fffR51

jmank88 replied:
We still need to pass a real context here. How will this be integrated in to the rest of the system? Normally we implement a service and have StopCh to link to shutdown.

4of9 commented on core/services/telemetry/chip_ingress.go:
```diff
@@ -0,0 +1,152 @@
+// Used for migration from OTI to ChIP Ingress
+type ChipIngressAdapter struct {
+	Network       string
+	ChainID       string
```
is `ChainID` just an EVM concept, or is it used in other chain families as well?

kirqz23 replied:
It looks like for Solana the concept of `chainid` exists as well: https://github.com/smartcontractkit/chain-selectors/blob/main/solana.go#L27-L41

kirqz23 commented on core/services/telemetry/chip_ingress.go:
```diff
@@ -0,0 +1,152 @@
+	"fmt"
+	"strings"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
```
consider using more descriptive `chainselector`

pkcll committed:
address comments

pkcll committed:
Revert 'use context.WithTimeout in SendLog'

pkcll committed:
Merge branch 'develop' of github.com:smartcontractkit/chainlink into INFOPLAT-2731-oti-chip-ingress-chain-selector

patrickhuie19 commented on core/services/telemetry/chip_ingress.go:
```diff
@@ -0,0 +1,155 @@
+
+// SendLog implements commontypes.MonitoringEndpoint
+// It forwards the telemetry log to the beholder emitter with proper domain/entity attributes
+func (a *ChipIngressAdapter) SendLog(log []byte) {
```
Should we be passing a parent context through to the Emit call here?

pkcll replied:
No, see the comment below 
```
// Not need to use context.WithTimeout because Emit is async and uses context.WithoutCancel(ctx)
```
